arch/risc-v/src/common/riscv_exception_common.S

```diff
@@ -147,6 +155,10 @@ exception_common:
   REGLOAD    s0, REG_INT_CTX(sp) /* restore mstatus   */
   csrw       mstatus, s0
 
+#ifdef CONFIG_ARCH_FPU
```

Comment:
should we restore before mstatus?

Reply:
OK, FPU restore would not touch the mstatus, but it's closer to current mainline (restore FPU context first)

Reply:
Ordering should be:
1. riscv_restorefpu() <- this modifies the context, not the status register first
2. restore mstatus <- restore status register with the updated context

Reply:
Yes, the latest change restore the fpu context first and then restore mstatus.